ledger/ledger-api-tests/suites/src/main/scala/com/daml/ledger/api/testtool/suites/v1_dev/ExplicitDisclosureIT.scala

```diff
@@ -319,281 +340,244 @@ final class ExplicitDisclosureIT extends LedgerTestSuite {
+              .update(_.arguments := ProtoArguments.CreateArguments(Dummy(delegate).arguments)),
+          )
+      } yield ()
   })
 
   test(
     "EDDuplicates",
     "Submission is rejected on duplicate contract ids or key hashes",
-    allocate(Parties(2)),
+    allocate(SingleParty, SingleParty),
     enabled = _.explicitDisclosure,
-  )(implicit ec => { case Participants(Participant(ledger, owner, delegate)) =>
-    for {
-      testContext <- initializeTest(
-        ledger,
-        owner,
-        delegate,
-        filterTxBy(owner, template = byTemplate, interface = None),
-      )
+  )(implicit ec => {
+    case Participants(
+          Participant(ownerParticipant, owner),
+          Participant(delegateParticipant, delegate),
+        ) =>
+      for {
+        testContext <- initializeTest(
+          ownerParticipant = ownerParticipant,
+          delegateParticipant = delegateParticipant,
+          owner = owner,
+          delegate = delegate,
+          transactionFilter = filterTxBy(owner, template = byTemplate, interface = None),
+        )
 
-      // Exercise a choice with a disclosed contract
-      _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract)
+        // Exercise a choice with a disclosed contract
+        _ <- testContext.exerciseFetchDelegated(testContext.disclosedContract)
 
-      // Submission with disclosed contracts with the same contract id should be rejected
-      errorDuplicateContractId <- testContext
-        .exerciseFetchDelegated(
-          testContext.disclosedContract,
-          testContext.disclosedContract.update(
-            // Set distinct key hash
-            _.metadata.contractKeyHash.set(ByteString.EMPTY)
-          ),
+        // Submission with disclosed contracts with the same contract id should be rejected
+        errorDuplicateContractId <- testContext
+          .exerciseFetchDelegated(
+            testContext.disclosedContract,
+            testContext.disclosedContract.update(
+              // Set distinct key hash
+              _.metadata.contractKeyHash.set(ByteString.EMPTY)
+            ),
+          )
+          .mustFail("duplicate contract id")
+
+        // Submission with disclosed contracts with the same contract key hashes should be rejected
+        errorDuplicateKey <- testContext
+          .exerciseFetchDelegated(
+            testContext.disclosedContract,
+            testContext.disclosedContract.update(
+              // Set distinct contract id
+              _.modify(_.withContractId("00" * 32 + "ff"))
+            ),
+          )
+          .mustFail("duplicate key hash")
+      } yield {
+        assertGrpcError(
+          errorDuplicateContractId,
+          LedgerApiErrors.CommandExecution.Interpreter.InvalidArgumentInterpretationError,
+          // TODO ED: Ensure contractId inlined in error message
+          Some("Found duplicated contract IDs in submitted disclosed contracts for template"),
```

Comment:
I don't think we should assert on the detailed error messages

Reply:
Ideally we should be as precise as possible in test assertions. So where we can, why not?

Reply:
The error messages (the human readable part) or not part of the LAPI contract so we should be able to change them whenever we fill a need. However, by asserting on them in the conformance tests you make it part of compatibility tests and we loose the flexibility. Then, even changing typos could become a headache.

Reply:
Yeah I agree that the compatibility is a problem, but I would leave this one here as well since:
1. I introduced it a while ago (so compatibility already remembers it)
2. the testing strategy should be orthogonal to some backwards compatibility restriction. They are painful once in a while to change, but why drive the testing coverage based on this restriction. If it's painful enough, we should maybe remove the restriction. Maybe a way forward is to introduce versioned assertions?